src/cuttlefish_generator.erl

```diff
@@ -45,11 +45,54 @@ map(Schema, Config) ->
 
 map(Schema, Config, ConfFile) ->
     IncludeFile = lists:filter(fun({K, _}) -> K =:= include_file end, Config),
-    AllConfig = merge_include_conf(Config -- IncludeFile, IncludeFile, ConfFile),
+    IncludedConfig = merge_include_conf(Config -- IncludeFile, IncludeFile, ConfFile),
+    AllConfig = merge_env_conf(IncludedConfig, Schema),
     map_add_defaults(Schema, AllConfig).
 
+-spec (merge_env_conf(cuttlefish_conf:conf(), cuttlefish_schema:schema()) -> cuttlefish_conf:conf()).
+merge_env_conf(Config, Schema) ->
+    Envs = match_env(getenv(), Schema),
+    lists:ukeymerge(1, Envs, Config).
+
+-spec (getenv() -> cuttlefish_conf:conf()).
+getenv() ->
+    lists:map(fun (E) -> [Key, Value] = string:split(E, "=", leading), {Key, Value} end, os:getenv()).
+
+match_env(Envs, {_, Mappings, _}) ->
+    Default = lists:sort(match_default_env(Envs, Mappings)),
+    Override = lists:sort(match_override_env(Envs, Mappings)),
+    lists:ukeymerge(1, Override, Default).
+
+match_default_env(Envs, Mappings) ->
+    lists:filtermap(fun (KV) -> do_match_default_env(KV, Mappings) end, Envs).
+do_match_default_env({"EMQX_" ++ Key, Value}, Mappings) ->
```

Comment:
we should perhaps parameterise this prefix.?
e.g `os:getenv(CUTTLEFISH_ENV_OVERRIDE_PREFIX)`